src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,379 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Duke {
+
+    public static final int DONE_START = 5;
+    public static final int TODO_START = 5;
+    public static final int EVENT_START = 6;
+    public static final int DEADLINE_START = 9;
+    public static final int DELETE_START = 7;
+    public static final int NEW_TASK_INDEX = 6;
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static int maxTaskIndex = 0;
+
+    public static Scanner in = new Scanner(System.in);
+    public static String input;
+
+    public static String filepath = "data.txt";
+
+    public static void main(String[] args) {
+        try {
+            loadData(filepath);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+
+        welcomeMessage();
+
+
+        input = getInput(in);
+        Command command;
+
+        while (!isBye()) {
+            if (isList()) {
+                command = Command.LIST;
+            } else if (isDone()) {
+                command = Command.DONE;
+            } else if (isTodo()) {
+                command = Command.TODO;
+            } else if (isEvent()) {
+                command = Command.EVENT;
+            } else if (isDelete()) {
+                command = Command.DELETE;
+            } else if (isDeadline()) {
+                command = Command.DEADLINE;
+            } else {
+                command = Command.INVALID;
+            }
+
+            try {
+                executeCommand(input, command);
+            } catch (InvalidCommandException e) {
+                System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            } catch (EmptyInputException e) {
+                System.out.println("OOPS!!! The description of a new task cannot be empty.");
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("OOPS!!! You need to add time for new Event or Deadline with '/at' or '/by'!!");
+            } catch (InvalidEventTimeException e) {
+                System.out.println("OOPS!!! You need to add time for new Event with keyword '/at'!!");
+            } catch (InvalidDeadlineTimeException e) {
+                System.out.println("OOPS!!! You need to add time for new Deadline with keyword '/by'!!");
+            } catch (NumberFormatException e) {
+                System.out.println("OOPS!!! I don't recognise the number to process Task Index to be done or deleted!!");
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("OOPS!!! You need to add valid Task Index to be done or deleted!!");
+            }
+
+            saveData();
+
+            System.out.println();
+            input = getInput(in);
+        }
+
+
```

Comment:
This blank line does not seem as useful the previous, at line 74. Using blank lines to group statements together are good but not in this example. See [https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically).